script.js

```diff
@@ -28,3 +28,57 @@ contact.addEventListener('click', () => {
   logo.style.display = 'block';
   hamburger.style.display = 'block';
 });
+
+const myRegex = /[A-Z]/;
+const form = document.getElementById('personal-info');
+form.addEventListener('submit', (event) => {
+  const { email } = form.elements;
+  const emailAddress = email.value;
+  const errM = document.querySelector('#form-valid');
+  if (myRegex.test(emailAddress) === true) {
+    event.preventDefault();
+    const message = 'Email field should only have <strong>lowercase</strong> letters';
+    errM.innerHTML = message;
+  }
+});
+
+const firstName = document.querySelector('#fname');
+const lastName = document.querySelector('#Last-name');
+const emailAddress = document.querySelector('#email');
+const message = document.querySelector('#message');
+
+firstName.value = localStorage.getItem('FirstName');
+lastName.value = localStorage.getItem('LastName');
+emailAddress.value = localStorage.getItem('emailAddress');
+message.value = localStorage.getItem('message');
+
+const formDetails = {
+  firstName: firstName.value,
+  lastName: lastName.value,
+  emailAddress: emailAddress.value,
+  message: message.value,
+};
+console.log(formDetails)
+localStorage.setItem('formDetails', formDetails);
+
+firstName.addEventListener('change', (event) => {
+  const firstNameValue = event.target.value;
+  console.log(firstNameValue)
+  localStorage.setItem('FirstName', firstNameValue);
+});
+
+lastName.addEventListener('change', (event) => {
+  const lastNameValue = event.target.value;
+  localStorage.setItem('LastName', lastNameValue);
+});
+
+emailAddress.addEventListener('change', (event) => {
+  emailAddress.value = event.target.value;
+  const emailAddressValue = emailAddress.value;
+  localStorage.setItem('emailAddress', emailAddressValue);
+});
+
+message.addEventListener('change', (event) => {
+  const messageValue = event.target.value;
+  localStorage.setItem('message', messageValue);
+});
```

Comment:
- [ ] Good job by storing the form data for all inputs in a single object. So I noticed that they are saved separately in the local storage registry instead of being in a **single local storage registry** Make sure you fix that issue as it is required for the project.

- [ ] Kindly make sure once data are saved to the local storage the data object is turned into a string using `JSON.stringify` or similar methods as required for the project. You can watch this[ video to get an idea](https://www.youtube.com/watch?v=AUOzvFzdIk4). 

#### Your Data in local storage
![local_storage2](https://user-images.githubusercontent.com/83042742/174414773-c572e962-03dc-4091-8a57-283c48b2b28d.png)

#### Expation behavior
![local_storage1](https://user-images.githubusercontent.com/83042742/174414766-ee51a3aa-3eb1-4431-bdd2-b275c8c5ecc9.png)